*** DodontoFServer.org.rb	2014-03-08 16:36:10.000000000 +0900
--- DodontoFServer.rb	2014-03-10 23:27:15.758154000 +0900
***************
*** 35,41 ****
  if( $isFirstCgi )
    require 'cgiPatch_forFirstCgi'
  end
! 
  require "config.rb"
  
  begin
--- 35,41 ----
  if( $isFirstCgi )
    require 'cgiPatch_forFirstCgi'
  end
!   
  require "config.rb"
  
  begin
***************
*** 43,49 ****
  rescue Exception
  end
  
- 
  if( $loginCountFileFullPath.nil? )
    $loginCountFileFullPath = File.join($SAVE_DATA_DIR, 'saveData', $loginCountFile)
  end
--- 43,48 ----
***************
*** 73,79 ****
    end
  end
  
! 
  
  $saveFileNames = File.join($saveDataTempDir, 'saveFileNames.json');
  $imageUrlText = File.join($imageUploadDir, 'imageUrl.txt');
--- 72,81 ----
    end
  end
  
! if( $useOpenJTalk == true )
!   # TTS プラグインを読み込む
!   require 'open_jtalk.rb'
! end
  
  $saveFileNames = File.join($saveDataTempDir, 'saveFileNames.json');
  $imageUrlText = File.join($imageUploadDir, 'imageUrl.txt');
***************
*** 820,825 ****
--- 822,830 ----
        ['changeEffectsAll', hasNoReturn], 
        ['removeEffect', hasNoReturn], 
        ['changeImageTags', hasNoReturn], 
+       
+       ['createTTS', hasReturn],
+       ['deleteTTS', hasReturn],
      ]
      
      commands.each do |command, commandType|
***************
*** 895,901 ****
      when 'getLoginInfo'
        return getWebIfLoginInfo
      end
!     
      loginOnWebInterface
      
      case commandName
--- 900,913 ----
      when 'getLoginInfo'
        return getWebIfLoginInfo
      end
! 
!     case commandName
!     when 'createTTS'
!       return sendWebIfCreateTTS
!     when 'deleteTTS'
!       return sendWebIfDeleteTTS
!     end
!    
      loginOnWebInterface
      
      case commandName
***************
*** 920,926 ****
      when 'getLoginUserInfo'
        return getWebIfLoginUserInfo
      end
!     
      return {'result'=> "command [#{commandName}] is NOT found"}
    end
    
--- 932,938 ----
      when 'getLoginUserInfo'
        return getWebIfLoginUserInfo
      end
! 
      return {'result'=> "command [#{commandName}] is NOT found"}
    end
    
***************
*** 1498,1503 ****
--- 1510,1553 ----
      
      return result
    end
+ 
+   # WEBIFのコマンドパーサ(case 文) から呼ばれるメソッド
+   # createTTS 辺りの名前で case 文判定していただけると良いかと思います。
+   # DodontoFServer#analizeCommand メソッドへ
+   # commands ハッシュ へ追加で ['createTTS', hasReturn] というのが望ましいかと。
+   def sendWebIfCreateTTS
+     logging("sendWebIfCreateTTS begin")
+     saveData = {}
+     
+     input_file = getRequestData('input_file')
+     output_file = getRequestData('output_file')
+     # 追加ボイスは未実装です。
+     # voice_type = getRequestData('voice_type')
+     
+     # create TTS voice file.
+     logging("create TTS voice file: "+input_file)
+     saveData['result'] = open_jtalk(input_file, output_file)
+     logging("sendWebIfCreateTTS end")
+ 
+     return saveData
+   end
+   
+   # WEBIFのコマンドパーサ(case 文) から呼ばれるメソッド
+   # deleteTTS 辺りの名前で case 文判定していただけると良いかと思います。
+   # DodontoFServer#analizeCommand メソッドへ
+   # commands ハッシュ へ追加で ['deleteTTS', hasReturn] というのが望ましいかと。
+   def sendWebIfDeleteTTS
+     logging("sendWebIfDeleteTTS begin")
+     saveData = {}
+     
+     target_file = getRequestData('target_file')
+     logging("delete TTS voice file.")
+     FileUtils.remove_file(target_file) if FileTest.exists?(target_file)
+     saveData['result'] = !FileTest.exists?(target_file)
+     logging("sendWebIfDeleteTTS end")
+ 
+     return saveData
+   end
    
    def getHashValue(hash, key, default)
      value = hash[key]
***************
*** 6470,6476 ****
      #通常のテキストファイル形式
      main(cgiParams)
    end
-   
  end
    
  if( $0 === __FILE__ )
--- 6520,6525 ----
