*** DodontoFServer.org.rb	2013-12-24 22:32:54.000000000 +0900
--- DodontoFServer.rb	2014-03-11 19:52:30.531395945 +0900
***************
*** 57,62 ****
--- 57,66 ----
  # require "FileLock2.rb"
  require "saveDirInfo.rb"
  
+ if( $useOpenJTalk == true )
+   # TTS プラグインを読み込む
+   require 'open_jtalk.rb'
+ end
  
  $saveFileNames = File.join($saveDataTempDir, 'saveFileNames.json');
  $imageUrlText = File.join($imageUploadDir, 'imageUrl.txt');
***************
*** 811,816 ****
--- 815,823 ----
        ['changeEffectsAll', hasNoReturn], 
        ['removeEffect', hasNoReturn], 
        ['changeImageTags', hasNoReturn], 
+        
+       ['createTTS', hasReturn],
+       ['deleteTTS', hasReturn],
      ]
      
      commands.each do |command, commandType|
***************
*** 886,892 ****
      when 'getLoginInfo'
        return getWebIfLoginInfo
      end
!     
      loginOnWebInterface
      
      case commandName
--- 893,906 ----
      when 'getLoginInfo'
        return getWebIfLoginInfo
      end
!  
!     case commandName
!     when 'createTTS'
!       return sendWebIfCreateTTS
!     when 'deleteTTS'
!       return sendWebIfDeleteTTS
!     end
!      
      loginOnWebInterface
      
      case commandName
***************
*** 1482,1487 ****
--- 1496,1539 ----
      
      return result
    end
+ 
+   # WEBIFのコマンドパーサ(case 文) から呼ばれるメソッド
+   # createTTS 辺りの名前で case 文判定していただけると良いかと思います。
+   # DodontoFServer#analizeCommand メソッドへ
+   # commands ハッシュ へ追加で ['createTTS', hasReturn] というのが望ましいかと。
+   def sendWebIfCreateTTS
+     logging("sendWebIfCreateTTS begin")
+     saveData = {}
+     
+     input_file = getRequestData('input_file')
+     output_file = getRequestData('output_file')
+     # 追加ボイスは未実装です。
+     # voice_type = getRequestData('voice_type')
+     
+     # create TTS voice file.
+     logging("create TTS voice file: "+input_file)
+     saveData['result'] = open_jtalk(input_file, output_file)
+     logging("sendWebIfCreateTTS end")
+ 
+     return saveData
+   end
+   
+   # WEBIFのコマンドパーサ(case 文) から呼ばれるメソッド
+   # deleteTTS 辺りの名前で case 文判定していただけると良いかと思います。
+   # DodontoFServer#analizeCommand メソッドへ
+   # commands ハッシュ へ追加で ['deleteTTS', hasReturn] というのが望ましいかと。
+   def sendWebIfDeleteTTS
+     logging("sendWebIfDeleteTTS begin")
+     saveData = {}
+     
+     target_file = getRequestData('target_file')
+     logging("delete TTS voice file.")
+     FileUtils.remove_file(target_file) if FileTest.exists?(target_file)
+     saveData['result'] = !FileTest.exists?(target_file)
+     logging("sendWebIfDeleteTTS end")
+ 
+     return saveData
+   end
    
    def getHashValue(hash, key, default)
      value = hash[key]
